pyugrid/ugrid.py

```diff
@@ -116,68 +109,175 @@ def __init__(self,
 
         self.mesh_name = mesh_name
 
-        # the data associated with the grid
-        # should be a dict of UVar objects
-        self._data = {}  # The data associated with the grid.
+        # The data associated with the grid should be a dict of UVar objects.
+        self._data = {}
         if data is not None:
             for dataset in data.values():
                 self.add_data(dataset)
 
-        # A kdtree is used to locate nodes.
-        # It will be created if/when it is needed.
         self._kdtree = None
 
     @classmethod
-    def from_ncfile(klass, nc_url, mesh_name=None, load_data=False):
+    def load_grid(klass, nc, mesh_name=None, load_data=False):
         """
-        create a UGrid object from a netcdf file name (or opendap url)
+        Loads UGrid object from a `netCDF4.Dataset` object or file name/URL,
+        adding the data to the passed-in grid object.
 
-        :param nc_url: the filename or OpenDap url you want to load
+        It will load the mesh specified, or look for the first one it finds if
+        none is specified.
+
+        :param nc: The grid data to be loaded up.
+        :type nc: `netCDF4.Dataset` object or netCDF/URL string
+
+        :param mesh_name=None: name of the mesh to load.
+        :type mesh_name: string
 
-        :param mesh_name=None: the name of the mesh you want. If None, then
-                               you'll get the only mesh in the file. If there
-                               is more than one mesh in the file, a ValueError
-                               Will be raised
         :param load_data=False: flag to indicate whether you want to load the
```

Comment:
Hmm -- if the netcdf Variables are lazy loaded, maybe we don't need this at all. When I first wrote this, UVars turned the netcdf variable objects into numpy arrays right off -- you don't want that, but if you just keep a reference to the variables in a UVAR, then why not have them all loaded at once to be queried?